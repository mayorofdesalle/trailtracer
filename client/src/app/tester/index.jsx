import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import './mocks/i18n.mock';
import './mocks/matchMedia.mock';
import './mocks/resizeObserver.mock';

import store from '@app/store';
import GlobalStyle from '@components/style/GlobalStyle';
import Theme from '@components/style/Theme';

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                <GlobalStyle />
                <BrowserRouter>
                    {children}
                </BrowserRouter>
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };
