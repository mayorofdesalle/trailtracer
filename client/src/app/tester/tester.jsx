import { render } from '@testing-library/react'
import { Provider } from 'react-redux';

import './mocks/matchMedia.mock';
import store from '@app/store';
import Theme from '@components/style/Theme'

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                {children}
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };