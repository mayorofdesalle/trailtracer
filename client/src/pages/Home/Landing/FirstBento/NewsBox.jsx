import styled from 'styled-components';

import { slideFromBottom } from '@components/style/anims';
import { pseudoBentoArrowTop } from '@components/ui/Bento/Arrows';
import BentoBox from '@components/ui/Bento/BentoBox';

/**
 * NewsBox
 * @description
 * Styled bento box to contain the news content.
 **/
const NewsBox = styled(BentoBox)`
    grid-area: NewsBox;
    
    --arrowColor: ${({ theme }) => theme.colors.secondary};
    ${pseudoBentoArrowTop}

    &::before {
        animation: ${slideFromBottom} 1s ease-in-out;
    }
`;

export default NewsBox;