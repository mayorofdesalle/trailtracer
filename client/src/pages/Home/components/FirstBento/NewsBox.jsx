import styled from 'styled-components';

import BentoBox from '@components/ui/BentoBox';
import { glassMorph } from '@components/misc/mixins';
import { rotateInFB } from '@components/misc/anims';

import { pseudoBentoArrowTop } from '../mixins';

/**
 * NewsBox
 * @description
 * This is a styled bento box that is used to contain the news content.
 **/
const NewsBox = styled(BentoBox)`
    outline: 2px solid ${({ theme }) => theme.colors.primaryTransparent};
    animation: ${rotateInFB} 1s ease-in-out;

    --arrowColor: ${({ theme }) => theme.colors.secondary};
    ${pseudoBentoArrowTop}

    ${glassMorph}
`;

export default NewsBox;