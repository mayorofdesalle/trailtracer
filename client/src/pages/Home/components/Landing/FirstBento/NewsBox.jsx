import styled from 'styled-components';

import BentoBox from '@components/ui/Bento/Box';
import { pseudoBentoArrowTop } from '@components/misc/Mixins';
import { slideFromBottom } from '@components/misc/Anims';

/**
 * NewsBox
 * @description
 * This is a styled bento box that is used to contain the news content.
 **/
const NewsBox = styled(BentoBox)`
    grid-area: NewsBox;
    
    --arrowColor: ${({ theme }) => theme.colors.secondary};
    ${pseudoBentoArrowTop}

    &::before {
        animation: ${slideFromBottom} 1s ease-in-out;
    }
`;

export default NewsBox;