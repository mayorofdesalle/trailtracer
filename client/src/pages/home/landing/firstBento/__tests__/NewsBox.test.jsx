import { describe, expect, it, render, screen } from '@app/tester';

import NewsBox from '../NewsBox';

describe('NewsBox', () => {

    it('should render correctly', () => {
        render(<NewsBox data-testid='news-box' />);
        const newsBox = screen.getByTestId('news-box');

        expect(newsBox).toBeInTheDocument();
        expect(newsBox).toHaveStyle('grid-area: NewsBox');
    });

});