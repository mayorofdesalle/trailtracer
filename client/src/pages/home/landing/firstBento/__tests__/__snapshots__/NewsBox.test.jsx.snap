// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NewsBox > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-beySPh sc-guDLey sc-hLQSwg jVdZA lbUchL dNITVc"
  />
</DocumentFragment>
`;
